internal/gengapic/client_init.go

```diff
@@ -144,16 +144,18 @@ func (g *generator) internalClientIntfInit(serv *descriptor.ServiceDescriptorPro
 	return nil
 }
 
-// This is a helper function similar to methodDoc that includes a deprecation notice for deprecated services.
+// serviceDoc is a helper function similar to methodDoc that includes a deprecation notice for deprecated services.
 func (g *generator) serviceDoc(serv *descriptor.ServiceDescriptorProto) {
 	com := g.comments[serv]
 
-	// If the service is marked as deprecated and there is not a comment explaining it, then add default deprecation comment.
-	// If the service includes a deprecation notice, then use that.
+	// If the service is marked as deprecated and there is no comment explaining it, then add default deprecation comment.
+	// If the service includes a deprecation notice, then prepend a comment with the service name stating it is deprecated and use the included deprecation notice.
 
 	if serv.GetOptions().GetDeprecated() {
-		if com == "" || !strings.Contains(com, "Deprecated") {
-			com += "\n\nDeprecated: This may be removed in a future version."
+		if !containsDeprecated(com) {
+			com += "\n" + serv.GetName() + " is deprecated.\n\nDeprecated: This may be removed in a future version."
```

Comment:
Sorry, one other place we can be clear with the name. Can we use the name instead of "This"? So, "Foo may be removed ..."

Reply:
Makes sense - changed to reference the name instead of "This."